File: leetcode3/이진희/3079. Find the Sum of Encrypted Integers.java
class Solution {
    public int sumOfEncryptedInt(int[] nums) {
        int sum = 0;

        for (int num : nums) sum += encrypt(num);
        
        return sum;
    }

    private int encrypt(int x) {
        int maxNum = 0;
        int len = 0;
        
        int tmp = x;
        while (tmp > 0) {
            int digit = tmp % 10;
            if (digit > maxNum) {
                maxNum = digit;
            }
            len++;
            tmp /= 10;
        }
        
        int ans = 0;
        for (int i = 0; i < len; i++) {
            ans = ans * 10 + maxNum;
        }
        
        return ans;
    }
}

File: leetcode3/최원준/3079. Find the Sum of Encrypted Integers.py
#

'''
1. 아이디어 :
-

2. 시간복잡도 :
    O(n * m)

3. 자료구조/알고리즘 :
-

'''
class Solution:
    def sumOfEncryptedInt(self, nums: List[int]) -> int:
        
        def encrypt(x):
            n = len(str(x))
            cmax = 0
            while x:
                remain = x%10
                cmax = max(cmax, remain)
                x = x//10
            return int(str(cmax) * n)

        return sum([encrypt(x) for x in nums])
